File: qqbot-client/src/main/java/com/mieai/qqbot/client/QqMarkdownMessageRequest.java
package com.mieai.qqbot.client;

import java.util.Map;
import java.util.Objects;
import java.util.Optional;

public record QqMarkdownMessageRequest(QqMessageTargetType targetType, String targetId, String content,
        Optional<String> customTemplateId, Map<String, String> params, Optional<String> replyMessageId,
        Optional<String> replyEventId, int messageSequence) {
    public QqMarkdownMessageRequest {
        Objects.requireNonNull(targetType, "targetType must not be null");
        if (targetId == null || targetId.isBlank()) throw new IllegalArgumentException("targetId is invalid");
        if (content == null || content.isBlank() || content.length() > 4000) throw new IllegalArgumentException("content is invalid");
        Objects.requireNonNull(customTemplateId, "customTemplateId must not be null");
        params = Map.copyOf(Objects.requireNonNull(params, "params must not be null"));
        Objects.requireNonNull(replyMessageId, "replyMessageId must not be null");
        Objects.requireNonNull(replyEventId, "replyEventId must not be null");
        if (messageSequence < 1) throw new IllegalArgumentException("messageSequence must be positive");
    }
}

File: qqbot-client/src/main/kotlin/com/mieai/qqbot/client/QqMarkdownMessageRequest.kt
package com.mieai.qqbot.client

class QqMarkdownMessageRequest(
    val targetType: QqMessageTargetType,
    val targetId: String,
    val content: String,
    val customTemplateId: String?,
    params: Map<String, String>,
    val replyMessageId: String?,
    val replyEventId: String?,
    val messageSequence: Int,
) {
    val params: Map<String, String> = params.toMap()

    init {
        require(targetId.isNotBlank()) { "targetId is invalid" }
        require(content.isNotBlank() && content.length <= 4000) { "content is invalid" }
        require(messageSequence >= 1) { "messageSequence must be positive" }
    }
}
